### PR TITLE
feat: Add contracts to compile flag to zksolcconfig

### PR DESCRIPTION
**Pull request by Jrigada**
# What :computer: 
* Added contracts_to_compile to Zksolcconfig and to TestArgs, BuildArgs and ScriptArgs

# Why :hand:
* Required to be able to compile the desired files when testing, building and scripting. 

# Evidence :camera:
1. `zkforge test --contracts-to-compile "Sign.t.sol"`

2. image

3. image

### CODE REVIEW

File: crates/zkforge/bin/cmd/test/mod.rs [hunk 1]
```diff
@@ -151,6 +155,7 @@ impl TestArgs {
         let mut project = config.project()?;
         // load the zkSolc config
         let mut zksolc_cfg = config.zk_solc_config().map_err(|e| eyre::eyre!(e))?;
+        zksolc_cfg.contracts_to_compile = self.contracts_to_compile.clone();
```
Comment (dutterbutter) [inline crates/zkforge/bin/cmd/test/mod.rs:158]: Shouldnt this be for builds, and scripts as well?

Comment (Jrigada) [reply crates/zkforge/bin/cmd/test/mod.rs:158]: Yes you are right sir, this should be replicated in build and scripts as well. Will do now 👍

**Comment by dutterbutter** [comment]
@Jrigada thinking about this more deeply. We should **also** provide the inverse of this. For example `--no-contracts-to-compile` which will compile everything but the specified contract(s). I think this is far more useful. For a practical example, compiling https://github.com/bgd-labs/aave-delivery-infrastructure will fail at first due to the use of EXTCODECOPY in the [`tests/PayloadScripts.t.sol#L142`](https://github.com/bgd-labs/aave-delivery-infrastructure/blob/main/tests/PayloadScripts.t.sol#L142). Ideally, users would then just run something like `zkforge zkbuild  --no-contracts-to-compile=PayloadScripts.t.sol`.

File: crates/zkforge/bin/cmd/script/mod.rs [hunk 2]
```diff
@@ -199,6 +199,14 @@ pub struct ScriptArgs {
...
     #[clap(flatten)]
     pub retry: RetryArgs,
+
+    /// Contracts to compile
+    #[clap(long, help_heading = "Contracts to compile", value_delimiter = ',')]
+    pub contracts_to_compile: Option<Vec<String>>,
+
+    /// Contracts to avoid compiling
+    #[clap(long, help_heading = "Contracts to avoid compilation", value_delimiter = ',')]
+    pub avoid_contracts: Option<Vec<String>>,
```
Comment (dutterbutter) [inline crates/zkforge/bin/cmd/script/mod.rs:209]: Perhaps move these to `CompilerArgs`?

**Comment by dutterbutter** [comment]
@Jrigada works great. Just the one comment.

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**